### PR TITLE
Add tests for %TypedArray%.prototype.fill

### PR DESCRIPTION
**Pull request by leobalter**
- [ ] ~~conversion operations~~
- [x] fill with other non number values that are not on the conversion operations
- [x] can I reach any abrupt completion (or observe) setting a value? [Step 7.b](https://tc39.github.io/ecma262/#sec-array.prototype.fill) (I'm asking to myself)

### CODE REVIEW

**Comment by leobalter** [comment]
I updated this PR to add the remaining tests. As the tests conversion operations are not merged yet, I'll apply the tests for `fill` on that PR.

File: test/built-ins/TypedArray/prototype/fill/fill-values-relative-start.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+  assert(
+    compareArray(new TA([0, 0, 0]).fill(8, 4), [0, 0, 0]),
+    'start position is never higher than this.length'
+  );
+
+  assert(
+    compareArray(new TA([0, 0, 0]).fill(8, -1), [0, 0, 8]),
+    'negative start sets initial position to max((this.length + relativeStart),0)'
+  );
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/fill/fill-values-relative-start.js]: This test should also assert the method's behavior when "len + relativeStart" is less than 0

File: test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  assert(
+    compareArray(new TA([0, 0, 0]).fill(8, 0, -1), [8, 8, 0]),
+    'negative end sets final position to max((this.length + relativeEnd), 0)'
+  );
+
+  assert(
+    compareArray(new TA([0, 0, 0]).fill(8, 0, 5), [8, 8, 8]),
+    'end position is never higher than of this.length'
+  );
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js]: This test should also assert the method's behavior when "len + relativeEnd" is less than 0

File: test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.fill
+es6id: 22.2.3.8
+description: >
+  Fills all the elements from a with a custom start index.
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js]: I think this should read, "with a custom *end* index"

**Comment by leobalter** [comment]
fixed at ab5ee50

File: test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+esid: sec-%typedarray%.prototype.fill
+es6id: 22.2.3.8
+description: >
+  Fills all the elements from a with a custom end index.
+info: >
+  22.2.3.8 %TypedArray%.prototype.fill (value [ , start [ , end ] ] )
+
+  %TypedArray%.prototype.fill is a distinct function that implements the same
+  algorithm as Array.prototype.fill as defined in 22.1.3.6 except that the this
+  object"s [[ArrayLength]] internal slot is accessed in place of performing a
```
Comment (jugglinmike) [inline test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js]: One-too-many quotation mark replacements :)

Comment (leobalter) [inline test/built-ins/TypedArray/prototype/fill/fill-values-relative-end.js]: 9e7bf77 oops

**Comment by jugglinmike** [comment]
Great! This looks good to me.

**Comment by jugglinmike** [comment]
Rebased, squashed and merged to `master` at commit 5e8b661050bf1d4395ad5760813db571f14991c7. Thanks, Leo!

**Pull request closed without merging**